pkg/productinfo/productinfo_test.go

```diff
@@ -473,3 +565,55 @@ func TestCachingProductInfo_GetRegions(t *testing.T) {
 		})
 	}
 }
+
+func TestCachingProductInfo_GetProductDetails(t *testing.T) {
+	tests := []struct {
+		name          string
+		ProductInfoer map[string]ProductInfoer
+		cache         Cache
+		checker       func(details []ProductDetails, err error)
+	}{
+		{
+			name: "successfully retrieved product details",
+			ProductInfoer: map[string]ProductInfoer{
+				"dummy": &DummyProductInfoer{},
+			},
+			cache: &DummyProductInfoer{TcId: ProductDetailsOK},
+			checker: func(details []ProductDetails, err error) {
+				assert.Nil(t, err, "the error should be nil")
+				assert.Equal(t, 3, len(details))
+			},
+		},
+		{
+			name: "successfully retrieved one product detail",
+			ProductInfoer: map[string]ProductInfoer{
+				"dummy": &DummyProductInfoer{},
+			},
+			cache: &DummyProductInfoer{TcId: GetProductDetail},
+			checker: func(details []ProductDetails, err error) {
+				assert.Nil(t, err, "the error should be nil")
+				assert.Equal(t, 1, len(details))
+				assert.Equal(t, []ProductDetails{{VmInfo: VmInfo{Type: "type-1", OnDemandPrice: 0.023,
```

Comment:
using equal for collections is hard to maintain; i'd suggest to assert on exact values (eg. the properties of the elements or elements of the collections instead)
this approach makes it easier to identify which value is wrong ...